### ISSUE TITLE
[DESIGN DOC]: New RestServer Architecture: RestServer -> DB -> ApiServer

### ISSUE THREAD

**Issue by yqwang-ms** [microsoft/pai#4651]
**By leveraging DB, RestServer can be**
1. RAW Consistency
2. High Perf and Powerful Query: List, Paging, Sorting, Summarizing, etc
3. Larger storage quota and duration
4. Active and history jobs are unified and merged together naturally (no need to introduce UID) (https://github.com/microsoft/pai/issues/3935)
5. Job Name can submit idempotently, attach metadata arbitrarily, and query uniquely (https://github.com/microsoft/pai/issues/3935)
6. etc

**SubTask**
https://github.com/microsoft/pai/issues/4650

**Features depend on it**
List History Jobs: https://github.com/microsoft/pai/issues/3845, https://github.com/microsoft/pai/issues/4610, https://github.com/microsoft/pai/issues/3935
Expose K8s events: part of enrich job debugging info: https://github.com/microsoft/pai/issues/4649

**New RestServer Architecture**
In short, compared with current architecture, we insert a DB between RestServer and ApiServer.
![image](https://user-images.githubusercontent.com/32826762/85936839-e9c82c00-b930-11ea-8ba6-f0809d7bd10b.png)

**Comment by fanyangCS** [comment]
relate to https://github.com/microsoft/pai/issues/4600
